core/trino-main/src/main/java/io/trino/connector/system/GlobalSystemConnector.java

```diff
@@ -93,11 +98,18 @@ public ConnectorSplitManager getSplitManager()
             @Override
             public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableFunctionHandle functionHandle)
             {
-                if (functionHandle instanceof SequenceFunctionHandle sequenceFunctionHandle) {
-                    return getSequenceFunctionSplitSource(sequenceFunctionHandle);
-                }
+                Set<TableFunction> tableFunctionAdditionalFeatures = tableFunctionProviders.stream()
+                        .map(factory -> factory.get(functionHandle))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(toImmutableSet());
 
-                throw new UnsupportedOperationException();
+                checkArgument(tableFunctionAdditionalFeatures.size() <= 1, "we should have only one implementation for concrete function handle or nothing");
```

Comment:
This should be a TrinoException with `FUNCTION_IMPLEMENTATION_ERROR` error code. Also, rephrase as "multiple implementations found for a function handle".